### PR TITLE
Automate Discord Notification for New Releases

### PR DESCRIPTION
**Pull request by adiraviraj**
## Description

This PR adds a new GitHub Action that automatically sends notifications to the IBC Gang Discord whenever a new release is tagged. The action uses the `Ilshidur/action-discord` GitHub Action and is triggered on the `release` event with `types: [published]`.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by DimitrisJim — APPROVED** [review]
lgmt! nice idea!

File: .github/workflows/discord_notify.yml
Comment (damiannolan) [inline .github/workflows/discord_notify.yml:1]: mega nit from me, but wanna edit the _ to a - in the file name so it aligns with other workflows?

**Review by damiannolan — APPROVED** [review]
Beautiful @adiraviraj, nice one! ❤️

**Review by srdtrk — APPROVED** [review]
lgtm!

**Pull request merged**